src/librustdoc/html/format.rs

```diff
@@ -141,15 +141,21 @@ crate struct WhereClause<'a> {
 }
 
 fn comma_sep<T: fmt::Display>(items: impl Iterator<Item = T>) -> impl fmt::Display {
-    display_fn(move |f| {
-        for (i, item) in items.enumerate() {
-            if i != 0 {
-                write!(f, ", ")?;
+    struct CommaSep<T: fmt::Display>(Vec<T>);
```

Comment:
I honestly prefer the cell over an allocation. Is collecting into a vector really necessary?

Reply:
I agree with @bugadani here.